### PR TITLE
Feature/cmn 650 material number on stacked and group modal items

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://jira.itkdev.dk/browse/CMN-650

#### Description

I've added material item number to the two items seen below aaand... 

#### Screenshot of the result

image

image

- Fee list now uses selectable material and the "new" group modal component. 
- Removed unused file
- Removed duplicated code
- Interpolated some string
- removed todo and added "AcceptTermsCheckbox.tsx" as this was duplicated (yay removing todos!)
 
#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

n/a

### CODE REVIEW

**Review by yepzdk — APPROVED** [review]

File: src/apps/fee-list/stackable-fees/AcceptTermsCheckbox.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <CheckBox
+      id="checkbox_id__fee_details"
+      onChecked={() => handleAcceptedTerms()}
+      label={
+        <>
+          {t("iAcceptText")}{" "}
+          <Link href={termsOfTradeUrl}>
+            {t("termsOfTradeText")}
+            <sup>*</sup>
```
Comment (spaceo) [inline src/apps/fee-list/stackable-fees/AcceptTermsCheckbox.tsx:27]: Wassup? :)

File: src/apps/fee-list/stackable-fees/fee-details-content.tsx [hunk 2]
```diff
@@ -34,78 +35,66 @@ const FeeDetailsContent: FC<FeeDetailsContentProps> = ({ feeDetailsData }) => {
...
+            {t("turnedInText", {
+              placeholders: { "@date": creationDateFormatted }
+            })}
+          </h2>
+        </div>
       </div>
+      <GroupModalContent
+        amountOfSelectableMaterials={0}
+        selectableMaterials={[]}
+        selectedMaterials={[]}
```
Comment (spaceo) [inline src/apps/fee-list/stackable-fees/fee-details-content.tsx:57]: Isnt it possible to define the empty arrays as default values on the props in the component?

File: src/apps/fee-list/stackable-fees/fee-details-content.tsx [hunk 3]
```diff
@@ -34,78 +35,66 @@ const FeeDetailsContent: FC<FeeDetailsContentProps> = ({ feeDetailsData }) => {
...
+                className="btn-primary btn-outline btn-small arrow__hover--right-small"
+                disabled={!showPaymentButton}
+              >
+                {t("payText")}
+              </button>
+            )}
+            {showPaymentButton && (
+              <Link
+                className="btn-primary btn-filled btn-small arrow__hover--right-small disabled"
+                href={new URL(paymentOverviewUrl)}
```
Comment (spaceo) [inline src/apps/fee-list/stackable-fees/fee-details-content.tsx:80]: If the url comes from the useUrl hook you do not need to create a new url object

File: src/apps/fee-list/stackable-fees/fee-details-content.tsx (hunk 2 above)
Comment (sinejespersen) [reply src/apps/fee-list/stackable-fees/fee-details-content.tsx]: done

File: src/apps/fee-list/stackable-fees/fee-details-content.tsx (hunk 3 above)
Comment (sinejespersen) [reply src/apps/fee-list/stackable-fees/fee-details-content.tsx]: done

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**